### PR TITLE
Allow disabling of GUI

### PR DESCRIPTION
**Pull request by mbtools**
Adding a parameter to GUI factory constructor allowing abapGit CI to disable GUI (since #6732 is not allowed to inject anymore).

### CODE REVIEW

**Comment by larshp** [comment]
for which use-case?

this possibly hooks into the steampunk enablement

or can it be the factory returning a different implementation, instead of adding complexity in the frontend class?

**Comment by mbtools** [comment]
it's here: https://github.com/abapGit/CI/commit/f079abc340e4a0ea9018c86d6ee322f9af5fc8c3

we could have the factory return a dummy implementation with GUI support, if that's better

File: src/ui/zcl_abapgit_ui_factory.clas.locals_imp.abap [hunk 1]
```diff
@@ -0,0 +1,51 @@
+CLASS lcl_frontend_services_no_gui DEFINITION.
```
Comment (larshp) [inline src/ui/zcl_abapgit_ui_factory.clas.locals_imp.abap:1]: just do another global class?

Comment (mbtools) [reply src/ui/zcl_abapgit_ui_factory.clas.locals_imp.abap:1]: `zcl_abapgit_frontend_no_gui` ?

Comment (larshp) [reply src/ui/zcl_abapgit_ui_factory.clas.locals_imp.abap:1]: yea

**Review by larshp — APPROVED** [review]
lgtm

**Pull request merged**